carbon-design-system/carbon-for-ibm-dotcom
# feat(masthead-v2): optimize megamenu tabs

### Related Ticket(s)

Resolves #10667 

### Description

Tabbed megamenus can have lots of content in them, creating a large DOM footprint and consequently a relatively heavy tax on device memory. This PR reduces this footprint by only rendering the active tab's tabpanel.

To facilitate this, we needed to refactor the way we "lazy-load" the megamenus so that they make use of Lit's rendering lifecycle. Previously they injected markup straight into the DOM, but this markup was somehow disconnected from Lit's reactivity mechanisms. As a result, updating a state property didn't successfully update the DOM content of the rendered megamenu.

### Changelog

**New**

- Only render the active tab's tabpanel in tabbed megamenus.

**Changed**

- Refactors masthead lazy-loading to use Lit's rendering lifecycle.

### To Test

Visit the [deploy preview](https://ibmdotcom-webcomponents.s3.us-east.cloud-object-storage.appdomain.cloud/deploy-previews/10815/index.html?path=/story/components-masthead--default) and verify the following with regards to tabbed megamenus:
- They render just the same as [the latest released version](https://carbon-design-system.github.io/carbon-for-ibm-dotcom/masthead-v2/web-components/?path=/story/components-masthead--default).
- Only the active tab's tabpanel (i.e. the right side of the megamenu) is present in the DOM.
